src/network/receiver.go

```diff
@@ -0,0 +1,35 @@
+package network
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+const BUFFER_SIZE = 1024
+
+/*
+Listen for incoming messages on specified IP and port.
+*/
+func ListenForMessages(ip string, port int, messageChannelchan chan<- []byte) {
+	conn, err := reuseport.ListenPacket("udp4", ip+":"+strconv.Itoa(port))
```

Comment:
Again, I suggest we change to `fmt.Sprintf("%s:%d", ip, port)`, I think it's cleaner. Can decide tomorrow.